File: kata-6kyu/catalog.js
function retrive(str, word){
  return str.substring(str.indexOf(word) + word.length + 1, str.lastIndexOf(word) - 2);
}
function catalog(s, article) {
  return s.split('\n')
        .filter(el=>el.includes(article))
        .map(el=> retrive(el, 'name') + ' > prx: $' + retrive(el, 'prx') + ' qty: ' + + retrive(el, 'qty'))
        .join('\r\n') || 'Nothing';
}

File: python-kata-6kyu/catalog.py
def catalog(s, article):
    arr = []
    s = s.replace('\n', '')
    s = s.replace('<prod><name>', '')
    s = s.replace('</name><prx>', ' > prx: $')
    s = s.replace('</prx><qty>', ' qty: ')
    s = s.split('</qty></prod>')
    for el in s:
        if article in el:
            arr.append(el)
    return '\r\n'.join(arr) or 'Nothing'
